### DISCUSSION TITLE
Use wl commands?

### DISCUSSION THREAD

**Discussion by francoism90** [francoism90/asus-router#1] [Ideas]
See [instructions](https://gist.github.com/francoism90/3dede7973354d067c41bff5e54203fe9?permalink_comment_id=4963750#gistcomment-4963750) given by @pasko-zh.

@pasko-zh I have tried your tweaks:

```bash
wl -i eth6 down
wl -i eth6 country ALL
wl -i eth6 country_abbrev_override "US"
wl -i eth6 regulatory 1
wl -i eth6 chanspec 36/160
wl -i eth6 up
wl -i eth6 5g_rate auto
wl -i eth6 txpwr1 -o -m 1496
```

The problem is that the interface/UI isn't in sync, and without running your commands, it still shows the correct nvram adjustments:

```bash
wl -i eth6 country (#a (#a/0) <unknown>)
wl -i eth6 chanspec (48/160 (0xeb32))
wl -i eth6 txpwr1 (TxPower is 127 qdbm, 31.75 dbm, 1496 mW  Override is Off)
```

**Comment by pasko-zh** [comment]
With `wl` command you modify the driver directly, i.e. in RAM, without changing anything in NVRAM. The GUI reads everything from NVRAM, therefore, you will only see what is stored there and not what you changed via `wl` command.

If one would like to persist `wl` command instructions, then a) put them in shell script and or b) put them in a startup script.

btw, was a good idea to move it from gist to github repo

**Reply by francoism90** [reply]
@pasko-zh Thanks for your reply. :)

I'm confused, since it's NVRAM passing changing to the wireless-interface(s). If you save in the GUI (or CLI), it should read and overwrite the current NVRAM parameters, and restart/reload the wireless service. By not using `nvram commit`, is should only overrule in memory, right?

Setting values using `wl` can also become an issue when you use AiMesh (like I do), since the NVRAM parameters aren't being synced.

I think you may need both of them, but my testing with `wl` overruling only, didn't work.

**Reply by ronzino83** [reply]
> With `wl` command you modify the driver directly, i.e. in RAM, without changing anything in NVRAM. The GUI reads everything from NVRAM, therefore, you will only see what is stored there and not what you changed via `wl` command.
> 
> If one would like to persist `wl` command instructions, then a) put them in shell script and or b) put them in a startup script.
> 
> btw, was a good idea to move it from gist to github repo

keep in mind that wl commands, modifies NVRAM without a commit. this means that RAM value is changed but not persistent after reboot. The GUI reads data from RAM, populated from NVRAM at reboot.

For example, if in a fresh device you sent this sequence via SSH...
wl -i eth7 down
wl -i eth7 country ALL
wl -i eth7 country_abbrev_override "US"
wl -i eth7 up

you will see in the GUI that more options raise up in drop down menus. So WL has changed RAM and GUI is reading RAM. If you reboot the router, you will go back to previous settings in GUI but also on hardware as HW setting come from NVRAM at reboot time, not WL commands.

The problem of broken GUI comes from here. If you look at the web page code Advanced_Wireless_Content.asp available under www folder of you router, you will see that menus and ticks are populated reading from RAM (nvram get) and actions are taken (nvram set) based on hard coded choice. So a code revision should be done in order to have a proper GUI behavior aligned with the unlocked potential of WL and NVRAM manual tuning

**Reply by pasko-zh** [reply]
Also note, that sometimes the following happens: You change and persist NVRAM, you reboot and then enter the GUI (you will see all channels for instance, i.e. the changes from NVRAM are reflected in the GUI) and then change something in the GUI, e.g., select a different channel, and then press apply. And then **päng:** Your settings, like tx power, channels, country settings etc., are back to the settings when you bought the router.... 
(those settings are stored in a ROM part of your router and you cannot change them... but the trick via NVRAM or WL command works quite well as we all know 😎)

So, this is one reason I prefer CLI to a GUI -- ok, my personal taste 😂

**Comment by pasko-zh** [comment]
Maybe we could add also a section in the [readme](https://github.com/francoism90/asus-router) how to work with `wl` commands? 
When I find time, I could do pull request.

**Reply by francoism90** [reply]
@pasko-zh That's perfectly fine, ty.

I was thinking of moving a few things to the Wiki, this will keep the README minimal.
